src/main/java/io/cdap/plugin/salesforce/connector/SalesforceConnector.java

```diff
@@ -0,0 +1,188 @@
+import io.cdap.cdap.etl.api.connector.PluginSpec;
+import io.cdap.cdap.etl.api.connector.SampleRequest;
+import io.cdap.cdap.etl.api.validation.ValidationException;
+import io.cdap.plugin.common.ConfigUtil;
+import io.cdap.plugin.salesforce.SObjectDescriptor;
+import io.cdap.plugin.salesforce.SalesforceConnectionUtil;
+import io.cdap.plugin.salesforce.SalesforceConstants;
+import io.cdap.plugin.salesforce.SalesforceSchemaUtil;
+import io.cdap.plugin.salesforce.authenticator.AuthenticatorCredentials;
+import io.cdap.plugin.salesforce.plugin.SalesforceConnectorConfig;
+import io.cdap.plugin.salesforce.plugin.sink.batch.SalesforceBatchSink;
+import io.cdap.plugin.salesforce.plugin.sink.batch.SalesforceSinkConfig;
+import io.cdap.plugin.salesforce.plugin.source.batch.MapToRecordTransformer;
+import io.cdap.plugin.salesforce.plugin.source.batch.SalesforceBatchSource;
+import io.cdap.plugin.salesforce.plugin.source.batch.SoapRecordToMapTransformer;
+import io.cdap.plugin.salesforce.plugin.source.batch.util.SalesforceSourceConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Salesforce Connector Plugin
+ */
+@Plugin(type = Connector.PLUGIN_TYPE)
+@Name(SalesforceConstants.PLUGIN_NAME)
+@Description("Connection to access data in Salesforce SObject.")
+public class SalesforceConnector implements DirectConnector {
+  private static final Logger LOG = LoggerFactory.getLogger(SalesforceConnector.class);
+  private static final String ENTITY_TYPE_OBJECTS = "object";
+  private static final String LABEL_NAME = "label";
+  private final SalesforceConnectorConfig config;
+  private StructuredRecord record;
+
+  SalesforceConnector(SalesforceConnectorConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void test(ConnectorContext connectorContext) throws ValidationException {
+    FailureCollector collector = connectorContext.getFailureCollector();
+    config.validate(collector);
+  }
+
+  @Override
+  public BrowseDetail browse(ConnectorContext connectorContext, BrowseRequest browseRequest) throws IOException {
+    AuthenticatorCredentials credentials = new AuthenticatorCredentials(config.getUsername(), config.getPassword(),
+                                                                        config.getConsumerKey(),
+                                                                        config.getConsumerSecret(),
+                                                                        config.getLoginUrl());
+    BrowseDetail.Builder browseDetailBuilder = BrowseDetail.builder();
+    int count = 0;
+    try {
+      PartnerConnection partnerConnection = SalesforceConnectionUtil.getPartnerConnection(credentials);
+      DescribeGlobalResult dgr = partnerConnection.describeGlobal();
+      // Loop through the array echoing the object names to the console
```

Comment:
fix comment: we are not echoing object names to console here

Reply:
Done